### PR TITLE
Require rsync on system

### PR DESCRIPTION
**Pull request by AJR-VMware**
Add utility function for detecting command on PATH.
Add checks for presence of rsync before its used.

Convert image used for building binaries in CI
Image used to build gpbackup go binaries was missing rsync.
Udpated to use a heavier build image.

### CODE REVIEW

File: utils/util.go [hunk 1]
```diff
@@ -140,9 +146,9 @@ func InitializeSignalHandler(cleanupFunc func(bool), procDesc string, termFlag *
 		fmt.Println() // Add newline after "^C" is printed
 		switch sig {
 		case unix.SIGINT:
-				gplog.Warn("Received an interrupt signal, aborting %s", procDesc)
+			gplog.Warn("Received an interrupt signal, aborting %s", procDesc)
 		case unix.SIGTERM:
-				gplog.Warn("Received a termination signal, aborting %s", procDesc)
+			gplog.Warn("Received a termination signal, aborting %s", procDesc)
```
Comment (bmdoil) [inline utils/util.go:151]: This seems unrelated.

File: utils/agent_remote.go [hunk 2]
```diff
@@ -40,6 +40,11 @@ func CreateSegmentPipeOnAllHosts(oid string, c *cluster.Cluster, fpInfo filepath
 }
 
 func WriteOidListToSegments(oidList []string, c *cluster.Cluster, fpInfo filepath.FilePathInfo) {
+	rsync_exists := CommandExists("rsync")
+	if !rsync_exists {
+		gplog.Fatal(errors.New("Failed to find rsync on PATH.  Please ensure rsync is installed."), "")
```
Comment (bmdoil) [inline utils/agent_remote.go:45]: nit: remove 2nd space before "Please"

**Review by bmdoil — COMMENTED** [review]
Can you updated the commit message to remove these lines since they no longer apply.
```
try removing autoreq disable

add requirement to ubuntu installer

Remove rsync from reqs
```

File: utils/util.go (hunk 1 above)
Comment (AJR-VMware) [reply utils/util.go:151]: I didn't do this one manually.  Must have been autoformat on save?  I'm not sure.  I'll remove this.  Good catch, thanks!

Comment (AJR-VMware) [reply utils/util.go]: Fixed!

File: utils/agent_remote.go (hunk 2 above)
Comment (AJR-VMware) [reply utils/agent_remote.go]: Fixed!

**Review by bmdoil — APPROVED** [review]
+1
